pyfr/integrators/dual/pseudo/multip.py

```diff
@@ -97,26 +98,30 @@ def ntotiters(iself):
                 def convmon(iself, *args, **kwargs):
                     pass
 
-                def finalise_pseudo_advance(iself, *args, **kwargs):
-                    pass
-
-                def _rhs_with_dts(iself, t, uin, fout):
+                def _rhs_with_dts(iself, t, uin, fout, mg_add=True):
                     # Compute -∇·f
                     iself.system.rhs(t, uin, fout)
 
-                    # Coefficient for the current solution state
-                    scoeff = iself.stepper_coeffs[0]/iself._dt
+                    if iself.stage_nregs > 1:
+                        iself._add(0, self._stage_regidx[iself.currstg],
+                                   1, fout)
+
+                    # Registers
+                    vals = iself.stepper_coeffs[:2] + [1]
+                    regs = [fout, iself._idxcurr, iself._source_regidx]
 
                     # Physical stepper source addition -∇·f - dQ/dt
-                    iself._add(1, fout, scoeff, iself._idxcurr,
-                               1, iself._source_regidx, subdims=iself._subdims)
+                    iself._add(*it.chain(*zip(vals, regs)),
```

Comment:
Why is chain needed here?

Reply:
*zip only produce tuples with two entries. chain and zip together is used in few other places like https://github.com/PyFR/PyFR/blob/develop/pyfr/integrators/dual/pseudo/pseudosteppers.py#L350